gcsio/src/test/java/com/google/cloud/hadoop/gcsio/EventLoggingHttpRequestInitializerTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.mockTransport;
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.cloud.hadoop.util.testing.MockHttpTransportHelper;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+public class EventLoggingHttpRequestInitializerTest {
+  private static final String requestUrl = "http://google.com";
+
+  private EventLoggingHttpRequestInitializer requestInitializer;
+  private EventLoggingHttpRequestInitializer requestInitializerSpy;
+  private ArgumentCaptor<HashMap<String, Object>> logDetailsCaptor;
```

Comment:
I think that Flogger/Java Util Logging should have some ways to test log records that do not require intercepting - if they exist then it could be better to use it to test this e2e.

Reply:
Done.